airflow/providers/amazon/aws/operators/emr.py

```diff
@@ -824,7 +824,7 @@ def execute(self, context: Context) -> str | None:
                 job_flow_id=self._job_flow_id,
                 log_uri=get_log_uri(emr_client=self._emr_hook.conn, job_flow_id=self._job_flow_id),
             )
-        if self.deferrable:
+        if self.deferrable and self.wait_for_completion:
```

Comment:
1. I'd consider refactoring both conditions as follows (a bit more readable IMO):
```python
if self.wait_for_completion:
    if self.deferrable:
       ...
    else:
       ...

``` 

2. Unit tests + docstring update for this behavior would be helpful.

Reply:
Is not it the case for 1? But I definitely agree on 2 :)